tests/SpecTests/DocumentsMatchConstraint.php

```diff
@@ -276,6 +276,10 @@ private function assertEquals(ArrayObject $expected, ArrayObject $actual, $ignor
         }
 
         foreach ($expected as $key => $expectedValue) {
+            if ($expectedValue === null) {
+                continue;
```

Comment:
I assume this was done to satisfy the [Fields that must NOT be present in Actual Documents](https://github.com/mongodb/specifications/commit/92255e0e57027646e8ef23399d3f33fdc114221f#diff-c4a2b694daca296938e31db2cd9b5206R289) spec test change introduced by mongodb/specifications#692.

I think this is technically incorrect, as it's skipping the assertion entirely if the expected value is `null`. At the very least, I think you'd want to assert that `$actualHasKey` (set below) is `false`.\\But rather than modify DocumentsMatchConstraint at all (which inadvertently affect other assertions, such as change stream results), I'd propose handling this in `CrudSpecTest::assertCommandMatches()`, similar to as is done in [TransactionSpecTest here](https://github.com/mongodb/mongo-php-library/blob/1.6.0/tests/SpecTests/TransactionsSpecTest.php#L107). I believe the transaction spec tests are the only other tests that use the same "null means unset" concept for command assertions.

Lastly, I believe this PR is also missing `find-allowdiskuse.json` from https://github.com/mongodb/specifications/commit/92255e0e57027646e8ef23399d3f33fdc114221f.

Reply:
Good point. I've moved the check to not affect other tests. The `find-allowdiskuse.json` test was added in #717, but didn't get executed because we're not testing against a 4.3 server. Since some tests currently fail on 4.3, I'll add this in a separate PR and hold off on merging this until we're properly testing this functionality.

Reply:
Ah, I totally forgot that was included in my PR.

Reply:
Yep, I forgot to push this change 😅